x-pack/plugins/cloud_security_posture/server/routes/setup_routes.ts

```diff
@@ -23,6 +38,102 @@ import { defineGetDetectionEngineAlertsStatus } from './detection_engine/get_det
 import { defineBulkActionCspBenchmarkRulesRoute } from './benchmark_rules/bulk_action/bulk_action';
 import { defineGetCspBenchmarkRulesStatesRoute } from './benchmark_rules/get_states/get_states';
 
+const getDataViewSafe = async (
+  soClient: ISavedObjectsRepository,
+  currentSpaceId: string,
+  currentSpaceDataViewId: string
+): Promise<SavedObject<DataViewAttributes> | undefined> => {
+  try {
```

Comment:
I use the `soClient` to retrieve data views and not the data view client because the data view client only returns data views for the default space.
More on that [here](https://elastic.slack.com/archives/C5TQ33ND8/p1721296310652019).